### PR TITLE
Set up example app

### PR DESCRIPTION
**Pull request by pepopowitz**
This PR sets up a demo app that combines parcel, react, typescript, and relay. 

As I started thinking about content for the relay-compiler exercise, I decided it would be easier to write if I actually had relay-compiler running in the project. A few shaved yaks later and I ended up hooking up the major dependencies we're planning on using for all exercises.

It uses the graphql version of the StarWars API as a data source. We've talked about using Artsy's API for these exercises but I vote to do that in a followup. 

## Does it `relay-compile`?

Yes:

image

## Does it type-check?

Yes:

image

## What's it look like?

image

## How'd it go?

TBH I'm pretty amazed I got it to work. I'll recap my adventure to demonstrate my "mastery": 

1. I started with [the Relay setup guide](https://relay.dev/docs/en/v2.0.0/installation-and-setup). I didn't finish it all the way before I got curious about combining this all with parcel. 
2. I ~"programmed computers"~ did a google search of `parcel react relay typescript` and found [this parcel issue that described problems configuring everything](https://github.com/parcel-bundler/parcel/issues/1590). It [links to a gist](https://gist.github.com/mathewbyrne/9d792f20da06cb4604c354091297dd70), which got me excited as it was a good representation of my desired end-state. 
3. I followed along with the gist and got things to the same point the issue described. `relay-compiler` successfully generated the files, and `tsc` successfully type-checked, but at run-time I got an error in my console -- which makes sense, because I followed a guide that literally told me I would get this error:
> Uncaught Invariant Violation: graphql: Unexpected invocation at runtime. Either the Babel transform was not set up, or it failed to identify this call site. Make sure it is being used verbatim as `graphql`. Note also that there cannot be a space between graphql and the backtick that follows.

Unfortunately, updating the `tsconfig.json` to contain `"jsx": "preserve"` [as one comment suggested](https://github.com/parcel-bundler/parcel/issues/1590#issuecomment-408097328) didn't get me past that error. I tried that exact `tsconfig.json` and still didn't get it to work - it's very possible I typo'ed or removed keys I didn't think I was removing.

4. I chased a red herring here and faffed around with .babelrc to see if the issue was that I just didn't have babel functioning (though I'm unclear how much of that is _just handled_ by parcel). This didn't get me anywhere.
5. I looked back at [the parcel docs](https://parceljs.org/typeScript.html#when-using-react) and they reassured me that I _should_ use `"jsx": "react"`, so I switched that back.
6. Then I looked at the two `tsconfig.json` examples in the above issue: [one from the linked gist](https://gist.github.com/mathewbyrne/9d792f20da06cb4604c354091297dd70#file-tsconfig-json), and [one from the comment](https://github.com/parcel-bundler/parcel/issues/1590#issuecomment-408097328). I diddled around with mine to combine those settings in different ways until...
7. It worked. `relay-compiler` successfully generates files, `tsc` successfully type-checks, and the app works. 🕺

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,232 @@
-/node_modules
\ No newline at end of file
+/node_modules# Logs
```
Comment (pepopowitz) [inline .gitignore:1]: I botched my commit message where I tried to explain this, but I created this .gitignore using `npx gitignore node`.

**Comment by zephraph** [comment]
I love this. Reading your process and journey is really helpful and informative. This is a _quality_ PR.

File: .gitignore (hunk 1 above)
Comment (zephraph) [reply .gitignore:1]: I had no idea that was a thing. How handy!

File: package.json [hunk 2]
```diff
@@ -8,15 +8,29 @@
...
   "private": true,
   "dependencies": {
+    "react": "^17.0.1",
+    "react-dom": "^17.0.1",
+    "react-relay": "^10.1.1",
+    "relay": "^0.8.0-1"
+  },
+  "scripts": {
+    "relay": "relay-compiler --src ./src --schema schema.graphql --extensions ts tsx --language typescript",
+    "start": "parcel src/index.html",
```
Comment (zephraph) [inline package.json:18]: This is why I really love prototyping things out w/ parcel. The build setup is just _so simple_.

**Pull request merged**